lib/api/model/model.dart

```diff
@@ -596,7 +596,10 @@ extension type const TopicName(String _value) {
 /// Different from [MessageDestination], this information comes from
 /// [getMessages] or [getEvents], identifying the conversation that contains a
 /// message.
-sealed class Conversation {}
+sealed class Conversation {
```

Comment:
```
api [nfc]: Extract Conversation.isSameAs

This will make it easier to support comparing the conversations
between subclasses of MessageBase.

The message list tests on displayRecipient are now mostly exercising the
logic on Conversation.isSameAs, which makes it reasonable to move the
tests.  Keep them here for now since this logic is more relevant to
message lists then it is to the rest of the app.
```

Commit-message nit: 'than it is'